util/codegen/codegen_test.go

```diff
@@ -174,3 +176,88 @@ func TestGenericContainsPointers(t *testing.T) {
 		})
 	}
 }
+
+func TestAssertStructUnchanged(t *testing.T) {
+	type args struct {
+		t      *types.Struct
+		tname  string
+		params *types.TypeParamList
+		ctx    string
+		it     *ImportTracker
+	}
+
+	// package t1 with a struct T1 with two fields
+	p1 := types.NewPackage("t1", "t1")
+	p1.Scope().Insert(types.NewTypeName(0, p1, "T1", types.NewStruct([]*types.Var{
+		types.NewField(0, nil, "P1", types.Typ[types.Int], false),
+		types.NewField(0, nil, "P2", types.Typ[types.String], false),
+	}, nil)))
+
+	// package t2 with a struct T2 with a named field T1 T1 and two other fields
+	p2 := types.NewPackage("t2", "t2")
+	p2.Scope().Insert(types.NewTypeName(0, p2, "T1", types.NewNamed(types.NewTypeName(0, p1, "T1", nil), nil, nil)))
+	p2.Scope().Insert(types.NewTypeName(0, p2, "T2", types.NewStruct([]*types.Var{
+		types.NewField(0, nil, "T1", types.NewNamed(types.NewTypeName(0, p1, "T1", nil), nil, nil), false),
+		types.NewField(0, nil, "P1", types.Typ[types.Int], false),
+		types.NewField(0, nil, "P2", types.Typ[types.String], false),
+	}, nil)))
+
+	// package t3 with a struct T3 with an embedded field T1 and two other fields
+	p3 := types.NewPackage("t3", "t3")
+	p3.Scope().Insert(types.NewTypeName(0, p3, "T1", types.NewNamed(types.NewTypeName(0, p1, "T1", nil), nil, nil)))
+	p3.Scope().Insert(types.NewTypeName(0, p3, "T2", types.NewStruct([]*types.Var{
+		types.NewField(0, nil, "T1", types.NewNamed(types.NewTypeName(0, p1, "T1", nil), nil, nil), true),
+		types.NewField(0, nil, "P1", types.Typ[types.Int], false),
+		types.NewField(0, nil, "P2", types.Typ[types.String], false),
+	}, nil)))
+
+	tests := []struct {
+		name string
+		args args
+		want []byte
+	}{
+		{
+			name: "t1-internally_defined",
+			args: args{
+				t:      p1.Scope().Lookup("T1").Type().Underlying().(*types.Struct),
```

Comment:
Use `t1` here?

Reply:
t1.Underlying().(*types.Struct) here, since t is `*types.Struct`, while t1 is `*types.Named`